### ISSUE TITLE
`crystal docs` does not work

### ISSUE THREAD

**Issue by caspiano** [PlaceOS/driver#83]
**Describe the bug**

`crystal docs` fails to run.

It looks like the compiler is somehow undergoing a second pass as constants and enums are redeclared (not allowed).

```
$ crystal docs --error-trace                                           
Showing last frame. Use --error-trace for full trace.

In src/driver/constants.cr:4:3

 4 | LOG_FORMATTER = ActionController.default_formatter
     ^------------
Error: already initialized constant PlaceOS::Driver::LOG_FORMATTER
```

**Expected behavior**

`crystal docs` should produce a docs folder.
This would be great as we'd be able to present docs on the framework surface in the body of the repository.

**Comment by caspiano** [comment]
Could potentially be resolved by #55

**Comment by stakach** [comment]
Testing shows that `crystal docs` works if we do the following (tested with private drivers repo)

1. convert drivers to stand-alone apps by adding `require "placeos-driver"`
1. convert specs to stand-alone apps by adding `require "placeos-driver/driver-specs/runner"`
1. remove `src/build.cr` by deleting the `src` folder
1. rename `drivers` to `src` (i.e. move all drivers to the `src` folder)
1. doc generation works

we'd need to be careful to maintain backwards compatibility - we can probably achieve this by:

1. leaving `build.cr` but having it do nothing unless env `COMPILE_DRIVER` is defined (so it doesn't error)
1. creating a symlink for the `drivers` folder that points to `src`

**Comment by stakach** [comment]
fixed in placeos/drivers@242a9c8a5aecb718bb753757d87fe373f7822e95
